### ISSUE TITLE
Initial Structures 

### ISSUE THREAD

**Issue by misslame** [misslame/Vampizza#4]
Be able to purchase, drag, place, and move two initial structures: Farm plot and Home.

**Comment by jamsge** [comment]
Vampizza should have drag and drop starting here: https://github.com/misslame/Vampizza/commit/1cc29e6d934ec4daedf323c32dc545ae01be4c14

Quick list of stuff I added in the drag drop merge for record keeping and information:
- Some grid changes. The grid now has three Tilemaps: (1) TilemapGround holds all ground related tiles, (2) TilemapStructure holds all structure related tiles, (3) TilemapPreview shows a preview of dragged structure. 
- Basic drag and drop functionality. Dragging structures from the inventory should close the menu, and give a preview of the structure as you drag it around. On mouse release, structure will be put down. TilemapStructure isn't changed until you let go of the mouse from a drag.
- Changed inventory repopulation logic. The first slot GameObject is no longer a child of SlotHolder. Removed some unnecessary destroys(), but I'm a little scared I didn't do this correctly and now there's some memory leak waiting to happen.

Shop purchases next!
